core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -822,18 +827,43 @@ private[spark] object MapOutputTracker extends Logging {
     } {
       objOut.close()
     }
-    val arr = out.toByteArray
+
+    val arr: Array[Byte] = {
+      val compressedOut = new ByteArrayOutputStream(4096)
+      val zos = new ZstdOutputStream(compressedOut)
+      Utils.tryWithSafeFinally {
+        compressedOut.write(DIRECT)
+        zos.write(out.getBuf, 0, out.size())
```

Comment:
I think we can replace this call with `out.writeTo(zos)`?  So that we can avoid exposing the `buf` directly.

Reply:
Thanks. Both of them has the same implementation, and `out.writeTo` is better.